config/config.go

```diff
@@ -149,6 +149,24 @@ func (c *C) ReloadConfig() {
 	}
 }
 
+func (c *C) ReloadConfigString(raw string) error {
```

Comment:
Should we have a lock in this method to make sure something doesn't double-trigger the reload, and a second one starts while the first is still running? Could that happen?